### PR TITLE
Add lint for foreign tables in API

### PR DESCRIPTION
**Pull request by olirice**
## What kind of change does this PR introduce?
Adds a lint to detect if foreign tables are on the API search path and accessible to `anon` or `authenticated`

### CODE REVIEW

File: lints/0016_materialized_view_in_api.sql [hunk 1]
```diff
@@ -6,7 +6,7 @@ select
     'WARN' as level,
     'EXTERNAL' as facing,
     array['SECURITY'] as categories,
-    'Detects materialized views that are potentially accessible over the Data APIs.' as description,
+    'Detects materialized views that are accessible over the Data APIs.' as description,
```
Comment (olirice) [inline lints/0016_materialized_view_in_api.sql:9]: unrelated change

File: docs/0017_foreign_table_in_api.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+Level: WARN
+
+### Rationale
+
+Foreign Tables in Postgres can present a security risk if they are accessible to API roles `anon` and `authenticated`. Unlike regular tables, foreign tables can not be configured to respect Row Level Security (RLS) policies.. Therefore, if foreign tables are accessible over APIs, all rows are always visible, which may not be intended.
```
Comment (inian) [inline docs/0017_foreign_table_in_api.md:5]:
```suggestion
Foreign Tables in Postgres can present a security risk if they are accessible to API roles `anon` and `authenticated`. Unlike regular tables, foreign tables can not be configured to respect Row Level Security (RLS) policies. Therefore, if foreign tables are accessible over APIs, all rows are always visible, which may not be intended.
```

File: docs/0017_foreign_table_in_api.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+You can test if your permissions update worked sucessfully by running
+
+```sql
+select pg_catalog.has_table_privilege('anon', 'public.some_foreign_table'::regclass::oid, 'select')
+-- Should return: 'false'
+```
+
+Substituting in the appropriate role and view name.
+
+If you do need to access data from the foreign table over APIs we recommend moving the foreign table out of the API's search path and then creating a function, accessible [over RPC](https://supabase.com/docs/reference/javascript/rpc), that implements security rules on top of the foreign table. For example, if we wanted to confirm that the Supabase Auth user matches the `author_id` column of the foreign table the function might look like:
```
Comment (inian) [inline docs/0017_foreign_table_in_api.md:26]:
```suggestion
If you do need to access data from the foreign table over APIs we recommend moving the foreign table out of the API's exposed schemas and then creating a function, accessible [over RPC](https://supabase.com/docs/reference/javascript/rpc), that implements security rules on top of the foreign table. For example, if we wanted to confirm that the Supabase Auth user matches the `author_id` column of the foreign table the function might look like:xt
```

File: docs/0017_foreign_table_in_api.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+```sql
+-- Create a new schema
+create schema private;
+
+-- Move the foreign table out of the API search path
```
Comment (inian) [inline docs/0017_foreign_table_in_api.md:32]:
```suggestion
-- Move the foreign table out of the API's exposed schemas
```

File: lints/0017_foreign_table_in_api.sql [hunk 5]
```diff
@@ -0,0 +1,43 @@
+create view lint."0017_foreign_table_in_api" as
+
+select
+    'foreign_table_in_api' as name,
+    'Foreign Table in API' as title,
+    'WARN' as level,
+    'EXTERNAL' as facing,
+    array['SECURITY'] as categories,
+    'Detects foreign tables that are accessible over APIs.' as description,
```
Comment (inian) [inline lints/0017_foreign_table_in_api.sql:9]:
```suggestion
    'Detects foreign tables that are accessible over APIs. Foreign tables do not respect row level security policies.' as description,
```

**Review by inian — APPROVED** [review]
